cmd/sdkstats/main.go

```diff
@@ -7,46 +7,141 @@ import (
-
 	"github.com/google/go-github/github"
 	"golang.org/x/oauth2"
 )
 
+type helmVersionParser struct{}
+type baseVersionParser struct {
+	searchQ      string
+	searchLatest string
+}
+type gomodVersionParser struct {
+	searchQ string
+}
+type unknownVersionParser struct{}
+
 func main() {
 	var token string
 	flag.StringVar(&token, "token", "", "GitHub API token")
 	flag.Parse()
-	args := flag.Args()
+	if len(token) == 0 {
+		log.Fatal("GITHUB API TOKEN MUST BE ENTERED \n Usage: './main --token=YOURTOKEN'")
+	}
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: args[0]},
+		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
-	client := github.NewClient(tc)
+	//client := github.NewClient(tc)
 
-	var queries []string
-
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/helm-operator")
-	queries = append(queries, "filename:Dockerfile quay.io/operator-framework/ansible-operator")
-	queries = append(queries, "filename:go.mod github.com/operator-framework/operator-sdk")
-	queries = append(queries, "filename:Gopkg.toml github.com/operator-framework/operator-sdk")
+	queries := []sdkstats.RepoMetadataQuery{
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "ansible",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/ansible-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ:      "quay.io/operator-framework/ansible-operator:v",
+				searchLatest: "quay.io/operator-framework/ansible-operator:master",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "helm",
+			Queries:     []string{"filename:Dockerfile quay.io/operator-framework/helm-operator"},
+			VersionParser: &baseVersionParser{
+				searchQ:      "quay.io/operator-framework/helm-operator:v",
+				searchLatest: "quay.io/operator-framework/helm-operator:latest",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType: "go.mod",
+			Queries: []string{
+				"filename:go.mod github.com/operator-framework/operator-sdk",
+			},
+			VersionParser: &gomodVersionParser{
+				searchQ: "replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType:   "gopkg.toml",
+			Queries:       []string{"filename:Gopkg.toml github.com/operator-framework/operator-sdk"},
+			VersionParser: &unknownVersionParser{},
+		},
+	}
+	// GetStats function for Query String from 'queries', These Strings are specific to Operator-SDK patterns.
+	collectStats := [][]sdkstats.RepoMetadata{}
 
 	for _, r := range queries {
-		stats, err := sdkstats.GetStats(client, r)
-		if len(stats) == 0 {
-			log.Fatal("GITHUB API is down at time, Please try after few minutes")
-		} else {
-			fileName := stats[0].ProjectType + ".json"
-			if err != nil {
-				fmt.Println(err)
+		stats, err := sdkstats.GetStats(ctx, tc, r)
+		if err != nil {
+			fmt.Printf("Failed to get stats for query  %q %v\n", r, err)
```

Comment:
`%q` requires a string, not a struct. Now that `r` is a struct and the queries is a slice of strings, it would be better to use `%v` and `r.Queries`. Also the go convention is to always separate the context from the underlying error with a colon:
```suggestion
			fmt.Printf("Failed to get stats for queries %v: %v\n", r.Queries, err)
```

Reply:
done